libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts

```diff
@@ -0,0 +1,87 @@
+// (C) 2020-2021 GoodData Corporation
+import { IInsightWidget } from "@gooddata/sdk-backend-spi";
+import { IAvailableDrillTargets } from "@gooddata/sdk-ui";
+import isEqual from "lodash/isEqual";
+import { useCallback, useState } from "react";
+import {
+    addDrillTargets,
+    IDrillTargets,
+    selectDrillTargetsByWidgetRef,
+    useDashboardDispatch,
+    useDashboardSelector,
+} from "../../../../model";
+
+export interface UseDashboardDrillTargetsProps {
+    widget: IInsightWidget;
+}
+
+interface UseDashboardDrillTargetsResult {
+    drillTargets: IDrillTargets | undefined;
+    onAvailableDrillTargetsReceived?: (availableDrillTargets?: IAvailableDrillTargets) => void;
+}
+
+/**
+ * @internal
+ */
+export const useDashboardDrillTargets = (
+    props: UseDashboardDrillTargetsProps,
+): UseDashboardDrillTargetsResult => {
+    const { widget } = props;
+
+    const dispatch = useDashboardDispatch();
+
+    const drillTargets = useDashboardSelector(selectDrillTargetsByWidgetRef(widget.ref));
+
+    const onAvailableDrillTargetsReceived = useCallback(
+        (availableDrillTargets?: IAvailableDrillTargets): void => {
+            if (
+                availableDrillTargets &&
+                !isEqual(drillTargets?.availableDrillTargets, availableDrillTargets)
+            ) {
+                dispatch(addDrillTargets(widget.ref, availableDrillTargets));
+            }
+        },
+        [widget, drillTargets],
+    );
+
+    return {
+        drillTargets,
```

Comment:
Do we want to return full drillTargets interface? Maybe availableDrillTargets is sufficient? Same question for the second hook.

Reply:
Done